gpu/segmentation/include/pcl/gpu/segmentation/impl/gpu_extract_clusters.hpp

```diff
@@ -60,17 +86,22 @@ pcl::gpu::extractEuclideanClusters (const pcl::PointCloud<pcl::PointXYZ>::Ptr  &
     max_answers = host_cloud_->size();
   else
     max_answers = max_pts_per_cluster;
-  std::cout << "Max_answers: " << max_answers << std::endl;
+  PCL_DEBUG("Max_answers: %i\n", max_answers);
 
   // to store the current cluster
   pcl::PointIndices r;
 
   DeviceArray<PointXYZ> queries_device_buffer;
   queries_device_buffer.create(max_answers);
 
+  // Host buffer for results
+  pcl::Indices sizes, data, cpu_tmp, gpu_indices;
```

Comment:
Similar to cpu_tmp, we can create a `gpu_tmp` here to replace the indices created in the new function on L54. That'll reduce more allocations for the GPU case (due to the loop at L101)

Reply:
As I outline below, I would suggest using either lambda and gpu_tmp or continue using the function and use gpu_tmp as a local variable.